File: Convert GIF to png.py
import imageio
import os
# 打开GIF文件
reader = imageio.get_reader(r'F:\dlc237.gif')

# 获取GIF中的帧数
num_frames = reader.get_length()

# 计算间隔
interval = num_frames // 108



# 指定保存图片的文件夹路径
save_path = r'D:\frames'

# 检查文件夹是否存在，如果不存在则创建
if not os.path.exists(save_path):
    os.makedirs(save_path)

# 提取并保存帧
for i in range(0, num_frames, interval):
    frame = reader.get_data(i)
    # 构建每个文件的完整路径
    filename = f'frame_{i//interval+1:03d}.png'
    filepath = os.path.join(save_path, filename)
    imageio.imwrite(filepath, frame)

# 关闭reader
reader.close()

File: convert GIF to png.R
# Installing and loading magick packages
install.packages("magick")
library(magick)

# Reading GIF files
gif_image <- image_read('F:\\dlc237.gif')

# Get GIF frames
num_frames <- image_info(gif_image)$frames

# Select every 100 frames
frames_to_save <- seq(1, num_frames, by = 100)

# Convert selected frames to individual images and save them
for (i in frames_to_save) {
  frame <- image_read(gif_image[i])
  frame <- image_convert(frame, format = 'png')
  image_write(frame, path = sprintf('frame_%03d.png', i))
}

